cpp/IRCCmds.cpp

```diff
@@ -226,3 +226,67 @@ void	KickCommand::execute(const std::string &params, IRCClient &client)
 
 	}
 }
+
+void	InviteCommand::execute(const std::string &params, IRCClient &client)
+{
+	std::istringstream	sstring(params);
+	std::string			nick, chanName;
+
+	sstring >> nick >> chanName;
+	if (sstring.fail())
+	{
+		client.sendMessage(ERR_NEEDMOREPARAMS(client.getNick(), "INVITE"));
+		return ;
+	}
+	if (!client.nickAlreadyInUse(nick, client.getFd()))
+	{
+		client.sendMessage(ERR_NOSUCHNICK(client.getNick(), nick));
+		return ;
+	}
+	if (!client.channelNameInUse(chanName))
+	{
+		client.sendMessage(ERR_NOSUCHCHANNEL(client.getNick(), chanName));
+		return ;
+	}
+	if (!client.getOp(chanName))
+	{
+		client.sendMessage(ERR_CHANOPRIVSNEEDED(client.getNick(), chanName));
+		return ;
+	}
+	client.getClient(nick).sendMessage(":" + client.getNick() + " INVITE " + nick + " " + chanName + "\r\n");
+}
+
+void TopicCommand::execute(const std::string &params, IRCClient &client)
+{
+    std::istringstream sstring(params);
+    std::string chanName, topic;
+
+    sstring >> chanName;
+    if (sstring.fail())
+    {
+        client.sendMessage(ERR_NEEDMOREPARAMS(client.getNick(), "TOPIC"));
+        return;
+    }
+    const IRCServer* server = client.getServer();
```

Comment:
Ici, je comprends la démarche, mais ce que tu dois faire, c'est compartimenter le code :

const IRCServer* server = client.getServer();
std::map<std::string, IRCChannel>::const_iterator it = server->getChannels().find(chanName);
if (it == server->getChannels().end())
{
client.sendMessage(ERR_NOSUCHCHANNEL(client.getNick(), chanName));
return;
}
Readapte-la pour qu’elle puisse être réutilisée, ça évitera la répétition quand on cherchera un channel avec le nom du chan.

Tu dois créer cette fonction dans la classe client : d'une, ça permettra de la réutiliser, et de deux, c'est que le code restera homogène, car c'est de cette manière qu'on la fait avec Sefir. De plus, tu pourras suprimer le getter sur serveur, car tu y auras accès dans la classe client.